packages/pwa-kit-dev/src/utils/resolver-utils.js

```diff
@@ -0,0 +1,132 @@
+    sourcePath = sourcePath.split('.')[0]
+    sourcePath = sourcePath.split(path.sep).join('/')
+    sourcePath = sourcePath.replace(indexRegExp, '')
+
+    // Do the same for the import path even thought it's not common to use /index and file extensions in your module
+    // imports.
+    importPath = importPath.split('.')[0]
+    importPath = importPath.replace(indexRegExp, '')
+
+    return sourcePath.endsWith(importPath)
+}
+
+/**
+ * Normalize and expand the extension configuration array so that it is easier to process.
+ * @param {{String, Object}[]} extensions - The extensions configuration value as defined in the PWA-Kit config.
+ * @returns {Object[]} extensions - The extensions array in object form.
+ *
+ * @example
+ * const result = expand(["store-finder", ["account-pages", {singlePage: true}], './extensions/local-extension']);
+ * console.log(result)
+ * // [["@salesforce/extension-store-finder", {}], ["@salesforce/extension-account-pages", {singlePage: true}], ["/home/project/extensions/local-extension", {}]]
+ */
+export const expand = (extensions = []) =>
+    extensions
+        .filter((extension) => Boolean(extension))
+        .map((extension) => {
+            let [nameRef, config = {}] = Array.isArray(extension) ? extension : [extension, {}]
+
+            switch (true) {
+                case /^\./.test(nameRef):
+                    // Relative Path
+                    nameRef = nameRef.split(/\/|\\/).join(path.sep)
+                    nameRef = path.join(process.cwd(), nameRef.replace('.', ''))
+                    break
+                case /^(\/|\\|\w:)/.test(nameRef):
+                    // Absolute Path (UNIX|Windows)
+                    nameRef = nameRef.split(/\/|\\/).join(path.sep)
+                    break
+                case /^@/.test(nameRef):
+                    // Do nothing
+                    break
+                default:
+                    // Default is to treat the reference as a extension "short" name.
+                    nameRef = `${EXTENSION_NAMESPACE}/${EXTENSION_PREFIX}-${nameRef}`
+                    break
+            }
+
+            return [nameRef, config]
+        })
+
+/**
+ * Based on the current extensibility configuration, return an array of candiate file paths to be used
+ * in the wild-card import module resolution for the given import path..
+ *
+ * @param {String} importPath - The import module-name.
+ * @param {String} sourcePath - The path the the file of the source import.
+ * @returns {String[]} paths - The potential paths to find the module import.
+ */
+export const buildCandidatePathArray = (importPath, sourcePath, opts = {}) => {
+    // Replace wildcard character as it has done its job getting us to this point.
+    importPath = importPath.replace('*/', '')
+
+    const {extensions = getConfig()?.app?.extensions} = opts
+    const isSelfReferenceImport = isSelfReference(importPath, sourcePath)
+    const cwd = process.cwd()
```

Comment:
I believe this is the current directory of which the Node.js process was started. This could give different unexpected results?

Maybe we could leverage the `projectDir` from the dev server

Reply:
Today, the webpack config is "static"  from the point of view of the dev server, but the dev server is in a good position to dynamically build the webpack config.

Reply:
Looking at some of the other definitions of projectDir they are simply cwd. But to add some flexibility to this util, I simply made the product dir an option that you can pass in with it's default to cwd. So going forward with the plugin work we can pass that value along like the extensions array.